backend/json_test.go

```diff
@@ -122,6 +122,14 @@ func TestResponseEncoder(t *testing.T) {
 			require.Equal(t, val.Error, other.Error)
 		}
 	})
+
+	t.Run("Unmarshalling a QueryDataResponse with an unexpected field should return an error", func(t *testing.T) {
```

Comment:
@ryantxu This is expected behaviour right? Just noticed this as we were thinking of those who might be using this functionality and try to unmarshal JSON with `status` and a previous version of the SDK